gui/src/app/book-list/book-list.component.ts

```diff
@@ -41,4 +43,24 @@ export class BookListComponent implements OnInit {
             }
         );
     }
+
+    searchBooks() {
+        if (this.title.trim() === '') {
+            this.ngOnInit();
+        } else {
+            this.bookService.searchBooks(this.title).subscribe(
+                (bookSearch: Book[]) => {
+                    this.books = bookSearch;
+                    this.errorMessage = '';
```

Comment:
errorMessage isn't used

use 
                this.snackBar.open(responseText, 'Close', {
                    duration: 15000, // Set the duration for which the message will be displayed
                });
                
                
to display it on the screen.